### ISSUE TITLE
Uneven results from Label Encoder problem

### ISSUE THREAD

**Issue by navcfc** [abhishekkrthakur/approachingalmost#15]
Hi,
I'm trying to run the code from the chapter `Approaching  Categorical Variables`. The code is run using XGBoost. I'm getting uneven results when I run the code. I'm not sure if I'm missing something. Can you help me look into this issue. 
This is the code that I used to run
`
import pandas as pd 
from sklearn import preprocessing
from sklearn import metrics

import xgboost as xgb

def run(fold):

    df = pd.read_csv('../input/train_folds.csv')

    num_cols = ['age', 'fnlwgt', 'education.num' , 'capital.gain', 'capital.loss' , 'hours.per.week']

    mapping = {
        '<=50K' : 0,
        '>50K' : 1
    }
    
    df.loc[:,'income'] = df['income'].map(mapping)
    features = [x for x in df.columns if x not in ('kfold','income') ]
    
    for col in features:
        if col not in num_cols:
            df.loc[:, col] = df[col].astype(str).fillna("NONE")

    for col in features:
        if col not in num_cols:
            lbl = preprocessing.LabelEncoder()

            lbl.fit(df[col])

            df.loc[:,col] = lbl.transform(df[col])
   

    df_train = df[df.kfold != fold].reset_index(drop = True)
    df_valid = df[df.kfold == fold].reset_index(drop = True)

    x_train = df_train[features].values
    x_valid = df_valid[features].values
    
    model = xgb.XGBClassifier(n_jobs = -1, max_depth=20, n_estimators= 200)

    model.fit(x_train, df_train.income.values)

    print("model trained")
    y_valid_preds = model.predict_proba(x_valid)[:,1]

    print(metrics.roc_auc_score(df_valid.income.values, y_valid_preds))

if __name__ == "__main__" :
    for f in range(5):
        run(f)
`

And the output that I'm getting is : 
`
fold 0: 0.7381942252913961
fold 1:0.031021225996466548
fold 2: 0.1365583437561918
fold 3: 0.47537328796809986
fold 4: 0.8349179819075687
`
The values being inconsistent are far away from the actual results.. I do not know what I'm doing wrong in here.
